### PR TITLE
ActivityIndicator Example ES6 classes migration and TimerMixin removal

### PR DESCRIPTION
**Pull request by artyomtrityak**
**This is the first part of `React.createClass` -> ES2015 classes migration.
1. Rewritten UI Explorer ActivityIndicator example to ES2015 classes
2. Removed TimerMixin from example.

Motivation:

- ES2015 classes do not support mixins so due to the classes / pure functions as "best practices" it would be better to avoid mixins in examples.

- TimerMixin is covered later and is out of scope of current example.

### CODE REVIEW

**Comment by facebook-github-bot-7** [comment]
By analyzing the blame information on this pull request, we identified @janicduplessis and @sebmarkbage to be potential reviewers.

**Comment by facebook-github-bot-7** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-1** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by artyomtrityak** [comment]
React Native repo ESLint passes with no errors, should i fix this errors from eslint-bot?

**Comment by janicduplessis** [comment]
Yes, I'm pretty sure those errors are coming from flow.

**Comment by artyomtrityak** [comment]
@janicduplessis true, thanks will fix

**Comment by artyomtrityak** [comment]
Are you expecting that Flowtype be a part of every UI Explorer React Native example? Once i'll start changing `React.createClass` to ES6 classes Flow will trow this type of errors.

**Comment by janicduplessis** [comment]
Yes, pretty much every example in UIExploerer uses flow.

**Comment by artyomtrityak** [comment]
@janicduplessis done

File: Examples/UIExplorer/ActivityIndicatorExample.js [hunk 1]
```diff
@@ -15,34 +15,43 @@
...
+ * Optional Flowtype state and timer types definition
+ */
+type StateT = { animating: boolean; };
+type TimerT = number;
+
+class ToggleAnimatingActivityIndicator extends Component {
+  /**
+   * Optional Flowtype state and timer types
+   */
+  state: StateT;
```
Comment (janicduplessis) [inline Examples/UIExplorer/ActivityIndicatorExample.js]: Nit: Can you just remove the `T` suffix from flow type names as we don't use this convention in the codebase.

Then this looks good to me!

**Comment by artyomtrityak** [comment]
done

**Comment by artyomtrityak** [comment]
any update here? I want to start rewriting UI Explorer examples to ES6 classes but want to merge this first

**Comment by facebook-github-bot-6** [comment]
@artyomtrityak I tried to find reviewers for this pull request and wanted to ping them to take another look. However, based on the blame information for the files in this pull request I couldn't find any reviewers. This sometimes happens when the files in the pull request are new or don't exist on master anymore. Is this pull request still relevant? If yes could you please rebase? In case you know who has context on this code feel free to mention them in a comment (one person is fine). Thanks for reading and hope you will continue contributing to the project.

**Comment by artyomtrityak** [comment]
rebased.
/cc @janicduplessis

**Comment by janicduplessis** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-0** [comment]
Thanks for importing.

**Pull request closed without merging**